src/program.ts

```diff
@@ -2991,13 +2990,28 @@ export abstract class DeclaredElement extends Element {
   get identifierAndSignatureRange(): Range {
     var declaration = this.declaration;
     var identifierNode = declaration.name;
-    if (declaration.kind == NodeKind.FUNCTIONDECLARATION || declaration.kind == NodeKind.METHODDECLARATION) {
+    var kind = declaration.kind;
+    if (kind == NodeKind.FUNCTIONDECLARATION || kind == NodeKind.METHODDECLARATION) {
       let signatureNode = (<FunctionDeclaration>declaration).signature;
       return Range.join(identifierNode.range, signatureNode.range);
     }
     return identifierNode.range;
   }
 
+  get parametersOrIdentifierRange(): Range {
+    var declaration = this.declaration;
+    var kind = declaration.kind;
+    if (kind == NodeKind.FUNCTIONDECLARATION || kind == NodeKind.METHODDECLARATION) {
+      let signatureNode = (<FunctionDeclaration>declaration).signature;
+      let parameters = signatureNode.parameters;
+      let numParams = parameters.length;
+      return numParams
+        ? Range.join(parameters[0].range, parameters[numParams - 1].range)
+        : signatureNode.range.atStart.extendBy(2);
```

Comment:
This seems brittle, since in source code there might well be `something ( )` with any number of whitespace. Could just use `signatureNode.range` I guess?

Reply:
Ideally, it would be nice to detect "(" and ")" in such cases

Reply:
These should be at `signatureNode.range.atStart` and `signatureNode.range.atEnd-1` I think, since a signature always starts at `(` and ends at `)`? Not 100% sure.